hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1136,10 +1217,18 @@ def add(self, data):
         if data.status > self.best_status:
             self.best_status = data.status
             self.reset()
-        self.examples.append(data)
+        self.fresh_examples.append(data)
```

Comment:
Because this is called unconditionally from `test_function`, I've been trying to figure out whether it's possible for the result added here to actually be a functional duplicate of a result that's already in the fresh or used list.

I think it is possible, and I don't think that's a problem in practice. But it might be worth noting that "fresh" examples aren't strictly guaranteed to actually be fresh.